conn.go

```diff
@@ -741,14 +756,19 @@ func (c *Conn) Handshake() Alert {
 		c.hState = state
 		logf(logTypeHandshake, "state is now %s", c.GetHsState())
 		_, connected = state.(StateConnected)
+		if connected {
+			c.state = state.(StateConnected)
+			c.handshakeComplete = true
```

Comment:
It seems like you could make this a little more compact:
```
if connectedState, connected := state.(StateConnected); connected {
    c.state = connectedState
    c.handshakeComplete = true
}
```

Reply:
That doesn't work. `connected` determines when the `for` exits.